src/mds/MDSDaemon.cc

```diff
@@ -1364,15 +1366,15 @@ bool MDSDaemon::ms_verify_authorizer(Connection *con, int peer_type,
       s->info.inst.addr = con->get_peer_addr();
       s->info.inst.name = n;
       dout(10) << " new session " << s << " for " << s->info.inst << " con " << con << dendl;
-      con->set_priv(s);
+      con->set_priv(RefCountedPtr{s, false});
       s->connection = con;
       if (mds_rank) {
         mds_rank->kick_waiters_for_any_client_connection();
       }
     } else {
       dout(10) << " existing session " << s << " for " << s->info.inst << " existing con " << s->connection
 	       << ", new/authorizing con " << con << dendl;
-      con->set_priv(s->get());
+      con->set_priv(RefCountedPtr{s, false});
```

Comment:
This causes the fs failures. This backport also needs #22384.

@vshankar

Reply:
thanks patrick -- I'll post the backport.

Reply:
follow-on luminous backport PR: https://github.com/ceph/ceph/pull/31049